app/collection.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+func (a *App) registerCollectionAndTopic(pluginID, collectionType, topicType string) error {
+	collectionTypeForPlugin, ok := a.ch.collectionTypes[pluginID]
+	if ok && collectionTypeForPlugin != collectionType {
+		return model.NewAppError("registerCollectionAndTopic", "app.collection.add_collection.exists.app_error", nil, "", http.StatusBadRequest)
+	} else if !ok {
+		a.ch.collectionTypes[pluginID] = collectionType
+	}
+
+	for _, ts := range a.ch.topicTypes {
+		if containsSlice(ts, topicType) {
+			return model.NewAppError("registerCollectionAndTopic", "app.collection.add_topic.exists.app_error", nil, "", http.StatusBadRequest)
```

Comment:
Should we ignore an error to register a duplicate collection and topic if it belongs to the same plugin?